weather@mockturtl/files/weather@mockturtl/applet.js

```diff
@@ -337,8 +337,28 @@ MyApplet.prototype = {
       Mainloop.timeout_add_seconds(3, Lang.bind(this, function mainloopTimeout() {
         this.refreshWeather(true)
       }))
+
+      this.orientation = orientation;
+      try {
+          this.setAllowedLayout(Applet.AllowedLayout.BOTH);
+          this.update_label_visible();
+      } catch(e) {
+          // vertical panel not supported
+      }
    }
 
+, update_label_visible: function () {
```

Comment:
I see two callsites for this function, both wrapped in a try-catch.  Is it possible to move the error handling inside it?